src/zimscraperlib/zim/creator.py

```diff
@@ -123,10 +127,14 @@ def start(self):
             self.set_mainpath(self.main_path)
 
         if getattr(self, "metadata", None):
-            self.update_metadata(**self.metadata)
+            self.__add_metadata(**self.metadata)
+
+        if not self.metadata.mandatory_values_all_set:
+            warnings.warn(f"{self.metadata.unset_keys} are unset.")
+        self.metadata.check_values_type()
         return self
 
-    def update_metadata(self, **kwargs):
```

Comment:
This breaks the API and would force us to release a new major version. Is this necessary?

Reply:
I would like talk about this part:
I think `update_metadata` is misleading here. update_metadata does not "update" a metadata, it actually adds a metadata. If you want to update an exist meta dataing, it will raise a RuntimeError at `python-libzim` level.
Here is a example:
Code:
```
with Creator(fpath, "welcome", "en") as creator:
    creator.add_item(StaticItem(path="welcome", content="hello"))
    creator.update_metadata(language="bam")
```
Error:
```
E   RuntimeError: Impossible to add M/Language
E     dirent's title to add is : Language
E     existing dirent's title is : Language

libzim/libzim.pyx:375: RuntimeError
RuntimeError: Impossible to add M/Language
```